third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -1184,6 +1260,162 @@ struct AtomicRMWOpConversion
     }
     return success();
   }
+
+private:
+  Value atomicIntraWaveReduce(PatternRewriter &rewriter, Value rmwPtr,
```

Comment:
Can we improve the documentation for this function?

* List the algorithm at the very beginning before code so we have an overall picture from get-go.
* For each code block, explain what block is for.